cmd/kops/create_cluster.go

```diff
@@ -1043,7 +1043,16 @@ func RunCreateCluster(f *util.Factory, out io.Writer, c *CreateClusterOptions) e
 	}
 	if cluster.Spec.API.LoadBalancer != nil && cluster.Spec.API.LoadBalancer.Type == "" {
 		switch c.APILoadBalancerType {
-		case "", "public":
+		case "":
+			if cluster.Spec.Topology.Masters == api.TopologyPrivate {
+				cluster.Spec.API.LoadBalancer.Type = api.LoadBalancerTypeInternal
+			} else {
+				cluster.Spec.API.LoadBalancer.Type = api.LoadBalancerTypePublic
+			}
+		case "public":
+			if cluster.Spec.Topology.Masters == api.TopologyPrivate {
+				return fmt.Errorf("Private topology supports --api-loadbalancer-type='internal' only")
```

Comment:
Hey @mikesplain, I'm not sure about this one. It should be possible to have such a setup (unless I've missed something here):
- AWS VPC with public & private subnets
- Master nodes with private topology
- API ELB on public subnets routing to master nodes on private subnets

Reply:
@KashifSaadat is correct I believe - you can have a public ELB with private topology.  It's our default for private topologies, I believe :-)

Reply:
@justinsb @KashifSaadat Yeah... I've been struggling to remember why I did that... I thought I tested something about subnets not being created or something but I can't seem to figure it out. Thanks!

Reply:
Ah, maybe you instead need to validate that Public subnets have been defined in the spec for this to be valid: `cluster.Spec.API.LoadBalancer.Type = api.LoadBalancerTypePublic` ?